File: 4.8-bc/ts/stage-3-hexagonal/src/notifications/infra/stub/sender.ts
// Заглушка Sender — логує у stdout. Для демо.
// Production-варіант жив би у notifications/infra/email/, notifications/infra/push/.
import type { Notification, Sender } from "../../domain/notification.js";

export class StubSender implements Sender {
  async send(n: Notification): Promise<void> {
    // eslint-disable-next-line no-console
    console.log(
      `[stub-sender] ${n.channel} -> user=${n.userId} payload=${JSON.stringify(n.payload)}`,
    );
    return Promise.resolve();
  }
}

File: 4.8-bc/py/stage-3-hexagonal/app/notifications/infra/stub/sender.py
"""Stub Sender — logs to stdout. Demo only.

Production-grade adapters live in ``infra/email/`` (SMTP), ``infra/push/``
(APNS/FCM), ``infra/sms/`` (Twilio etc).
"""
from __future__ import annotations

import logging

from app.notifications.domain.notification import Notification

logger = logging.getLogger(__name__)


class StubSender:
    async def send(self, n: Notification) -> None:
        logger.info(
            "[stub-sender] %s -> user=%s payload=%r",
            n.channel,
            n.user_id,
            n.payload,
        )
